### PR TITLE
Make HugePages const name consistent.

### PR DESCRIPTION
**Pull request by xiangpengzhao**
**What this PR does / why we need it**:
Make HugePages const name in v1 be consistent with the name in api.
https://github.com/kubernetes/kubernetes/blob/f9e6142f925a33a9d8a252ab79b9dabbee596ea2/pkg/api/types.go#L686

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/api/core/v1/types.go [hunk 1]
```diff
@@ -989,7 +989,7 @@ type StorageMedium string
 const (
 	StorageMediumDefault   StorageMedium = ""          // use whatever the default is for the node
 	StorageMediumMemory    StorageMedium = "Memory"    // use memory (tmpfs)
-	StorageMediumHugepages StorageMedium = "HugePages" // use hugepages
+	StorageMediumHugePages StorageMedium = "HugePages" // use hugepages
```
Comment (dixudx) [inline staging/src/k8s.io/api/core/v1/types.go]: Simply a rename, StorageMediumHugepages-> StorageMediumHugePages, right?

Comment (xiangpengzhao) [reply staging/src/k8s.io/api/core/v1/types.go]: exactly :)

Comment (jsafrane) [reply staging/src/k8s.io/api/core/v1/types.go]: Doesn't it break API compatibility? Do we guarantee any in `k8s.i/api`?

**Comment by jsafrane** [comment]
/retest
"Error checking queue status"

File: staging/src/k8s.io/api/core/v1/types.go (hunk 1 above)
Comment (xiangpengzhao) [reply staging/src/k8s.io/api/core/v1/types.go]: I don't think it breaks API compatibility. It just change the const name. It doesn't change the actual name.

Comment (jsafrane) [reply staging/src/k8s.io/api/core/v1/types.go]: Const name *is* API. Changing its name breaks builds to anyone who imported `k8s.i/api` and used `api.StorageMediumHugepages` in his go code. Question is if we support `k8s.i/api` stability.

@kubernetes/sig-api-machinery-pr-reviews

Comment (sttts) [reply staging/src/k8s.io/api/core/v1/types.go]: We don't. In major updates we have a number of changes usually which breaks compilation of 3rdparty code.

**Comment by sttts** [comment]
/assign @thockin 

For approval.

**Comment by lavalamp** [comment]
Removing the sig/api-machinery label: SIG API Machinery doesn't own the *contents* of the API.

/sig node

**Pull request merged**